forte/orbital-helpers/orbital_embedding.cc

```diff
@@ -834,37 +885,84 @@ std::shared_ptr<MOSpaceInfo> make_embedding(psi::SharedWavefunction ref_wfn,
     // Write a new MOSpaceInfo:
     std::map<std::string, std::vector<size_t>> mo_space_map;
 
-    // Frozen docc space
-    size_t freeze_o =
-        static_cast<size_t>(num_Fo + num_Bo + adj_sys_docc); // Add the additional frozen core to Bo
-    mo_space_map["FROZEN_DOCC"] = {freeze_o};
+    // Write orbital information into new mo_space_info based on different embedding methods:
 
-    // Restricted docc space
-    size_t ro = static_cast<size_t>(num_Ao - adj_sys_docc);
-    if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
-        ro -= diff;
-    }
-    mo_space_map["RESTRICTED_DOCC"] = {ro};
+    // ASET(mf):
+    if (options->get_str("EMBEDDING_TYPE") == "ASET_MF") {
+        // Normal partition for Frozen-Core embedding
+        size_t freeze_o = static_cast<size_t>(num_Fo + num_Bo +
+                                              adj_sys_docc); // Add the additional frozen core to Bo
+        mo_space_map["FROZEN_DOCC"] = {freeze_o};
 
-    // Active space
-    size_t a = static_cast<size_t>(actv_a[0]);
-    if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
-        a += diff;
-        a += diff2;
+        size_t ro = static_cast<size_t>(num_Ao - adj_sys_docc);
+        if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
+            ro -= diff;
+        }
+        mo_space_map["RESTRICTED_DOCC"] = {ro};
+
+        size_t a = static_cast<size_t>(actv_a[0]);
+        if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
+            a += diff;
+            a += diff2;
+        }
+        mo_space_map["ACTIVE"] = {a};
+
+        size_t rv = static_cast<size_t>(num_Av - adj_sys_uocc);
+        if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
+            rv -= diff2;
+        }
+        mo_space_map["RESTRICTED_UOCC"] = {rv};
+
+        size_t freeze_v = static_cast<size_t>(
+            num_Fv + num_Bv + adj_sys_uocc); // Add the additional frozen virtual to Bv
+        mo_space_map["FROZEN_UOCC"] = {freeze_v};

```

Comment:
Like I commented before, these part could be
```
bool hf_ref = options->get_str("EMBEDDING_REFERENCE") == "HF";

size_t ro = hf_ref ? static_cast<size_t>(num_Ao - adj_sys_docc) : static_cast<size_t>(num_Ao - adj_sys_docc - diff);
...
```
If you thi

Reply:
Thanks! I think this applies to many of my codes, I will revise them soon.